public/app/features/alerting/unified/mocks/server/handlers/datasources.ts

```diff
@@ -1,15 +1,32 @@
 import { http, HttpResponse } from 'msw';
 
+import { buildInfoResponse } from 'app/features/alerting/unified/testSetup/featureDiscovery';
+
 /** UID of the alertmanager that is expected to be broken in tests */
 export const MOCK_DATASOURCE_UID_BROKEN_ALERTMANAGER = 'FwkfQfEmYlAthB';
 /** Display name of the alertmanager that is expected to be broken in tests */
 export const MOCK_DATASOURCE_NAME_BROKEN_ALERTMANAGER = 'broken alertmanager';
 export const MOCK_DATASOURCE_EXTERNAL_VANILLA_ALERTMANAGER_UID = 'vanilla-alertmanager';
 export const MOCK_DATASOURCE_PROVISIONED_MIMIR_ALERTMANAGER_UID = 'provisioned-alertmanager';
+export const MOCK_DATASOURCE_GRAFANA_MIMIR = 'grafana-mimir';
+
+const isSupportedType = (uid: string): uid is keyof typeof buildInfoResponse => {
+  return uid in buildInfoResponse;
+};
 
 // TODO: Add more accurate endpoint responses as tests require
 export const datasourceBuildInfoHandler = () =>
-  http.get('/api/datasources/proxy/uid/:datasourceUid/api/v1/status/buildinfo', () => HttpResponse.json({}));
+  http.get<{ datasourceUid: keyof typeof buildInfoResponse | string }>(
+    '/api/datasources/proxy/uid/:datasourceUid/api/v1/status/buildinfo',
+    ({ params }) => {
+      const { datasourceUid } = params;
+      if (isSupportedType(datasourceUid)) {
+        const response = buildInfoResponse[datasourceUid];
+        return HttpResponse.json(response);
+      }
+      return HttpResponse.json({});
```

Comment:
Shouldn't we return 404?

Reply:
Ideally yes, but the existing test logic is working under the assumption of all datasources returning an empty object buildinfo. We should, over time, add more supported mock datasources and generalise the pattern that we have, but I haven't aimed to do that here :+1: But the endpoint should definitely eventually return 404's when we haven't defined a mock datasource